packages/plugin-react/src/index.ts

```diff
@@ -124,10 +125,23 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
     enforce: 'pre',
     config() {
       if (opts.jsxRuntime === 'classic') {
-        return {
-          esbuild: {
-            jsx: 'transform',
-          },
+        if ('rolldownVersion' in vite) {
+          return {
+            oxc: {
+              jsx: {
+                runtime: 'classic',
+                // disable __self and __source injection even in dev
+                // as this plugin injects them by babel
```

Comment:
Adding this because the initial reason was not obvious (compare to esbuild that ignores it)
```suggestion
                // as this plugin injects them by babel and oxc will throw
                //  if development is enabled and those properties are already present
```